components/applications-service/pkg/storage/postgres/services.go

```diff
@@ -238,6 +262,21 @@ func (db *Postgres) getServiceFromUniqueFields(name, member string) (*service, b
 	return &svc, true
 }
 
+// converts an array of composedService to an array of storage.Service
+func convertComposedServicesToStorage(svcs []*composedService) []*storage.Service {
```

Comment:
Wouldn't it be nice if there was a way to avoid this? There isn't. 😄

Reply:
well the way I thought it in the first place is that since both structs could be similar, by adding the `db:""` tags you could just re-use it. But I did cross a boundary there so, this is what you get to do the things right. 😅

Reply:
Yup, I was complaining about the language, not your code. 😉

Reply:
I didn't take it to the code but the language as you said, I just tried to work around the limitation but, wasn't a good idea 😂